modules/p2/ding_entity_rating/ding_entity_rating.module

```diff
@@ -0,0 +1,473 @@
+<?php
+
+/**
+ * @file
+ * Ding entity rating module.
+ *
+ * User rating of ting_objects with storage in OpenList
+ */
+
+include_once 'ding_entity_rating.serendipity.inc';
+
+// Ding rating list.
+define('DING_ENTITY_RATING_LIST', 'ratings');
+
+/**
+ * Implements hook_ctools_plugin_directory().
+ */
+function ding_entity_rating_ctools_plugin_directory($owner, $plugin_type) {
+  if ($owner == 'ctools' && $plugin_type == 'content_types') {
+    return 'plugins/content_types';
+  }
+}
+
+/**
+ * Implements hook_menu().
+ */
+function ding_entity_rating_menu() {
+  $items = array();
+  
+  // Add to list callback.
+  $items['ding_entity_rating/%ctools_js/%ting_object/%user/%'] = array(
+    'title' => 'Ding entity rating add to list',
+    'page callback' => 'ding_entity_rating_add_to_list_callback',
+    'page arguments' => array(1, 2, 3, 4),
+    'access arguments' => array('use personalisation'),
+  );
+  
+  $items['ding_entity_rating/get'] = array(
+    'title' => 'Ding entity get rating',
+    'page callback' => 'ding_entity_rating_get_user_rating_callback',
+    // 'page arguments' => array(2),
+    'access arguments' => array('use personalisation'),
+  );
+
+  return $items;
+}
+
+/**
+ * Menu hook callback for getting rating.
+ */
+function ding_entity_rating_get_user_rating_callback() {
+  $ratings = array();
+  
+  foreach (ding_entity_load_multiple($_GET['ids']) as $object) {
+    $ratings[$object->ding_entity_id] = ding_entity_rating_get_user_rating($object);
+  }
+  
+  echo drupal_json_encode($ratings);
+  drupal_exit();
+}
+
+/**
+ * Add to list callback.
+ *
+ * @param bool $ajax
+ *   If it's an ajax call or not.
+ * @param object $entity
+ *   The ting object to rate.
+ * @param object $account
+ *   The user account.
+ * @param int $rating
+ *   The given rating / weight.
+ */
+function ding_entity_rating_add_to_list_callback($ajax, $entity, $account, $rating) {
+  // Access check for permission.
+  global $user;
+  if ($user->uid != $account->uid) {
+    return '';
+  }
```

Comment:
@martinb14: This question still stands.

If there is a technical reason why it should not be in an access callback then a code comment explaining why would be good.